source/common/upstream/ring_hash_lb.cc

```diff
@@ -139,9 +138,13 @@ void RingHashLoadBalancer::Ring::create(
 #endif
 }
 
-void RingHashLoadBalancer::refresh() {
-  all_hosts_ring_.create(config_, host_set_.hosts());
-  healthy_hosts_ring_.create(config_, host_set_.healthyHosts());
+void RingHashLoadBalancer::refresh(uint32_t priority) {
+  while (per_priority_state_.size() < priority + 1) {
+    per_priority_state_.push_back(PerPriorityStatePtr{new PerPriorityState});
```

Comment:
Are gaps in the priority space expected? This would allocate a PerPriorityState for them even if it would never be used.

Reply:
Nope, it's documented in the EDS config that is not allowed

Reply:
We check this and fail the config, right? Do we have a test for that? I can't remember.

Reply:
I removed the config-fail.  I added a test for the locality (EdsTest.NoPriorityForLocalCluster) but not for the ring_hash check since I was planning on removing it the next day :-)

Reply:
Sorry what is the behavior if a management server provides a config with gaps?

Reply:
We create an empty host set for any gap levels and carry on.

Reply:
OK SGTM.